FHEM/00_SIGNALduino.pm

```diff
@@ -1085,14 +1135,32 @@ sub SIGNALduino_parseResponse($$$)
 sub
 SIGNALduino_ResetDevice($)
 {
-  my ($hash) = @_;
-  my $name = $hash->{NAME};
-
-  $hash->{logMethod}->($name, 3, "$name reset"); 
-  DevIo_CloseDev($hash);
-  my $ret = DevIo_OpenDev($hash, 0, "SIGNALduino_DoInit", 'SIGNALduino_Connect');
-
-  return $ret;
+	my $hash = shift;
+	my $name = $hash->{NAME};
+	if (!defined($hash->{helper}{resetInProgress}))
+	{
+		my $hardware = AttrVal($name,"hardware","");
+		$hash->{logMethod}->($name, 3, "$name/reset: $hardware"); 
+		DevIo_CloseDev($hash);
+	 	if ($hardware eq "radinoCC1101" && $^O eq 'linux') {
+			# The reset is triggered when the Micro's virtual (CDC) serial / COM port is opened at 1200 baud and then closed.
+			# When this happens, the processor will reset, breaking the USB connection to the computer (meaning that the virtual serial / COM port will disappear).
+			# After the processor resets, the bootloader starts, remaining active for about 8 seconds.
+			# The bootloader can also be initiated by pressing the reset button on the Micro.
+			# Note that when the board first powers up, it will jump straight to the user sketch, if present, rather than initiating the bootloader.	
+			my ($dev, $baudrate) = split("@", $hash->{DeviceName});
+			$hash->{logMethod}->($name, 3, "$hash->{TYPE} $name/reset: forcing special reset for $hardware on $dev");
+			# Mit dem Linux-Kommando 'stty' die Port-Einstellungen setzen
+			system("stty -F $dev ospeed 1200 ispeed 1200");
+			$hash->{helper}{resetInProgress}=1;
+			InternalTimer(gettimeofday()+1,"SIGNALduino_ResetDevice",$hash);
```

Comment:
Könntest Du testen, ob der Reset über die Device-by-id Variante funktioniert, wenn Du hier +7 Sekunden angibst?

Reply:
Das funktioniert natürlich auch:

2019.10.11 17:14:21 3: sduinoACM/reset: radinoCC1101
2019.10.11 17:14:21 3: SIGNALduino sduinoACM/reset: forcing special reset for radinoCC1101 on /dev/serial/by-id/usb-Unknown_radino_CC1101-if00
2019.10.11 17:14:21 3: sduinoACM/reset: reopen delayed for 7 second
2019.10.11 17:14:28 3: Opening sduinoACM device /dev/serial/by-id/usb-Unknown_radino_CC1101-if00
2019.10.11 17:14:28 1: sduinoACM: Can't open /dev/serial/by-id/usb-Unknown_radino_CC1101-if00: No such file or directory
2019.10.11 17:14:31 3: Setting sduinoACM serial parameters to 57600,8,N,1
2019.10.11 17:14:31 1: sduinoACM/define: /dev/serial/by-id/usb-Unknown_radino_CC1101-if00@57600
2019.10.11 17:14:31 1: sduinoACM/init: /dev/serial/by-id/usb-Unknown_radino_CC1101-if00@57600
2019.10.11 17:14:31 1: /dev/serial/by-id/usb-Unknown_radino_CC1101-if00 reappeared (sduinoACM)
2019.10.11 17:14:33 3: sduinoACM/init: disable receiver (XQ)
2019.10.11 17:14:33 3: sduinoACM/init: get version, retry = 0
2019.10.11 17:14:33 2: sduinoACM: initialized. v3.4.1_dev_09.09
2019.10.11 17:14:33 3: sduinoACM/init: enable receiver (XE)

Ich habe jetzt gleich die Verzögerung auf 10 Sekunden gesetzt, das vermeidet die Fehlermeldung:
2019.10.11 17:14:28 1: sduinoACM: Can't open /dev/serial/by-id/usb-Unknown_radino_CC1101-if00: No such file or directory

Das Log sieht dann so aus:
2019.10.11 17:23:09 3: sduinoACM/reset: radinoCC1101
2019.10.11 17:23:09 3: SIGNALduino sduinoACM/reset: forcing special reset for radinoCC1101 on /dev/serial/by-id/usb-Unknown_radino_CC1101-if00
2019.10.11 17:23:09 3: sduinoACM/reset: reopen delayed for 10 second
2019.10.11 17:23:19 3: Opening sduinoACM device /dev/serial/by-id/usb-Unknown_radino_CC1101-if00
2019.10.11 17:23:19 3: Setting sduinoACM serial parameters to 57600,8,N,1
2019.10.11 17:23:19 1: sduinoACM/define: /dev/serial/by-id/usb-Unknown_radino_CC1101-if00@57600
2019.10.11 17:23:19 1: sduinoACM/init: /dev/serial/by-id/usb-Unknown_radino_CC1101-if00@57600
2019.10.11 17:23:19 3: sduinoACM device opened
2019.10.11 17:23:20 3: sduinoACM/init: disable receiver (XQ)
2019.10.11 17:23:21 3: sduinoACM/init: get version, retry = 0
2019.10.11 17:23:21 2: sduinoACM: initialized. v3.4.1_dev_09.09
2019.10.11 17:23:21 3: sduinoACM/init: enable receiver (XE)